### PR TITLE
Support cases where entry server node is not yet available on entry st…

### PR DESCRIPTION
**Pull request by yossipapi**
…artup

### CODE REVIEW

File: lib/BackendClient.js [hunk 1]
```diff
@@ -252,6 +252,43 @@ var BackendClient = (function(){
         return deferred.promise;
     }
 
+    function getLiveEntryServerNodeId(entryId, serverType) {
```
Comment (kalturaguy) [inline lib/BackendClient.js]: why by serverType?  why not by the serverId?

File: lib/manifest/PlaylistGenerator.js [hunk 2]
```diff
@@ -53,6 +54,13 @@ function checkTranscodingProfileChange() {
         });
 }
 
+function updateStreamInfo() {
+    backendClient.updateStreamInfo(this.entryId, this.entryServerNodeId, this.flavorsObjArray)
```
Comment (kalturaguy) [inline lib/manifest/PlaylistGenerator.js]: forgot to add return

Comment (yossipapi) [inline lib/manifest/PlaylistGenerator.js]: Added

File: lib/BackendClient.js (hunk 1 above)
Comment (yossipapi) [inline lib/BackendClient.js]: What is the difference? Why is using one better than the other?

File: lib/manifest/PlaylistGenerator.js [hunk 3]
```diff
@@ -73,10 +81,16 @@ function savePlaylist() {
         .then(function() {
             that.lastPlaylistUpdate = new Date();
             that.logger.debug("Playlist written to disk");
-            backendClient.updateStreamInfo(that.entryId, that.entryServerNodeId, that.flavorsObjArray)
-                .then(function() {
-                    that.logger.debug("Flavors updated in DB");
-                });
+            if(that.entryServerNodeId) {
```
Comment (kalturaguy) [inline lib/manifest/PlaylistGenerator.js]: is this nicer?
 var getLiveEntryServerNodeIdPromise=that.entryServerNodeId ? Promise.resolve(that.entryServerNodeId) :  backendClient.getLiveEntryServerNodeId(that.entryId, that.entryServerType);

getLiveEntryServerNodeIdPromise.then(function(entryServerNodeId) {
                         that.entryServerNodeId = entryServerNodeId;
                         updateStreamInfo.call(that);
                     });

Comment (yossipapi) [inline lib/manifest/PlaylistGenerator.js]: Yep looks better for the Promise lovers among us :)
I will update the code accordingly.
👍

File: lib/manifest/PlaylistGenerator.js [hunk 4]
```diff
@@ -73,10 +83,16 @@ function savePlaylist() {
         .then(function() {
             that.lastPlaylistUpdate = new Date();
             that.logger.debug("Playlist written to disk");
-            backendClient.updateStreamInfo(that.entryId, that.entryServerNodeId, that.flavorsObjArray)
-                .then(function() {
-                    that.logger.debug("Flavors updated in DB");
-                });
+
+            var getLiveEntryServerNodeIdPromise =
+                that.entryServerNodeId ? Q.Promise.resolve(that.entryServerNodeId)
```
Comment (kalturaguy) [inline lib/manifest/PlaylistGenerator.js]: does this work?  i know only Q.resolve not Q.Promise.resolve

**Pull request merged**